### PR TITLE
Pivot table fixes

### PR DESCRIPTION
**Pull request by paulrosenzweig**
This PR bundles a handful of small fixes. The logically separate changes are in isolated commits. These descriptions are slightly better than the commit messages:

- [commit](https://github.com/metabase/metabase/commit/4f6506b6b74bc1762b1a111aadfdccbf9cd87598) Prevent hitting the pivot API when the query is missing a breakout. If you have an invalid query and switch to pivot tables we should display the error message without hitting the server.
- [commit](https://github.com/metabase/metabase/commit/ef7a28b0f2d4893df28d0742f264278c08f7e77a) Takes a first pass at differentiating subtotals with styling. Kyle is doing a better job branching off of this.
- [commit](https://github.com/metabase/metabase/commit/e2af980b473540a2e76fc79017f4ce9c83b5baa7) Fixes an issue where the "Totals for [VALUE]" didn't format [VALUE] at all. e.g. "Totals for 2020-01-01T00:00:00"
- [commit](https://github.com/metabase/metabase/commit/65582845680bc827b1d01cd4ecd8cc87a7cbe696) Fixes a bug where the "rows totals" columns would merge values together
- [commit](https://github.com/metabase/metabase/commit/7fdcf5dac9e49328aeb92c704399b0f85d499fcb) Small refactor to pair the logic that pulls apart the returned data with the general pivoting logic
- [commit](https://github.com/metabase/metabase/commit/daecf37b43fb95f5bf43b1306064f1ba04aae75b) Fixes and unskips pivot table unit tests
- [commit](https://github.com/metabase/metabase/commit/c7f6a60a3c568a3d41cfc7ccfcc3e48ad3ab7fd0) Wires up the public endpoint
- [commit](https://github.com/metabase/metabase/pull/14044/commits/80496bb1143671e88217e9c0aac6d01568748082) Avoid exception if you try to open pivot tables on native questions
- [commit](https://github.com/metabase/metabase/pull/14044/commits/5ebfedbfbcbd0949c9d366b2da983cfa90cfe8c6) Hide blue outline on pivot table clicks
- [commit](https://github.com/metabase/metabase/pull/14044/commits/7ca46b1863723d5e4f3d951deb2ffedf0a2389d8) Hide "table toggle" in QB for pivot tables

### CODE REVIEW

File: frontend/src/metabase/lib/data_grid.js [hunk 1]
```diff
@@ -87,8 +87,8 @@ function createRowSectionGetter({
 }) {
   const formatValues = values =>
     values === undefined
-      ? new Array(valueFormatters.length).fill(null)
-      : values.map((v, i) => valueFormatters[i](v));
+      ? new Array(valueFormatters.length).fill({ value: null })
```
Comment (ariya) [inline frontend/src/metabase/lib/data_grid.js:123]: Nitpick (we probably ought to standardize on this?): no need for `new`.

File: frontend/src/metabase/visualizations/visualizations/PivotTable.jsx [hunk 2]
```diff
@@ -165,30 +168,32 @@ export default class PivotTable extends Component {
       rowCount,
       columnCount,
     } = pivoted;
-    const cellWidth = 100;
-    const cellHeight = 25;
+    const CELL_WIDTH = 100;
+    const CELL_HEIGHT = 25;
```
Comment (ariya) [inline frontend/src/metabase/visualizations/visualizations/PivotTable.jsx:157]: Any reasons these consts are defined again?

**Review by ariya — APPROVED** [review]
LGTM :+1:

File: frontend/src/metabase/lib/data_grid.js (hunk 1 above)
Comment (paulrosenzweig) [reply frontend/src/metabase/lib/data_grid.js]: Oh neat I didn't realize that worked! I'm good either way. Without `new` is more concise, but with `new` is consistent with `Set` 🤷 . I updated it here.

File: frontend/src/metabase/visualizations/visualizations/PivotTable.jsx (hunk 2 above)
Comment (paulrosenzweig) [reply frontend/src/metabase/visualizations/visualizations/PivotTable.jsx]: Nope! I removed it.

**Pull request merged**